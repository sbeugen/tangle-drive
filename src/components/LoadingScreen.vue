<template>
  <div class="spinner-background" v-if="showLoader">
    <pulse-spinner :color="color"></pulse-spinner>
    <br>
    <h2>{{ getUploadText }}</h2>
  </div>
</template>

<script>
  import { mapActions, mapGetters } from 'vuex' 
  import PulseSpinner from 'vue-spinner/src/PulseLoader.vue'
  
  export default {
    components: {
      PulseSpinner
    },
    data() {
      return {
        color: '#87ceeb'
      }
    },
    methods: {
      
    },
    computed: {
      showLoader() {
        return !this.getFileUploadFinished || !this.getPowFinished
      },
      ...mapGetters('upload', [
        'getUploadText',
        'getFileUploadFinished',
        'getPowFinished'
      ])
    }
  }
</script>

<style>
  .spinner-background {
    height: 100%;
    width: 100%;
    background-color: rgba(0,0,0,0.7);
    padding-top: 50px;
    justify-content: center;
    position: fixed;
    top: 0;
    left: 0;
    z-index: 10000;
  }

</style>
