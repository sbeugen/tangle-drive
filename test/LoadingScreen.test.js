import { shallow, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'

import LoadingScreen from '../src/components/LoadingScreen'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('LoadingScreen.test.js', () => {
  let store
  let uploadGetters
  let downloadGetters
  let uploadActions

  beforeEach(() => {
    uploadGetters = {
      getUploadText: jest.fn().mockImplementation(() => 'Upload Prep.'),
      getFileUploadFinished: jest.fn().mockImplementation(() => false),
      getPowFinished: jest.fn().mockImplementation(() => true)
    }
    downloadGetters = {
      getDownloadText: jest.fn().mockImplementation(() => ''),
      getDownloadPreparationFinished: jest.fn().mockImplementation(() => true)
    }
    uploadActions = {
      setUploadText: jest.fn()
    }

    store = new Vuex.Store({
      strict: true,
      modules: {
        upload: {
          namespaced: true,
          getters: uploadGetters,
          actions: uploadActions
        },
        download: {
          namespaced: true,
          getters: downloadGetters
        }
      }
    })
  })

  describe('snapshot', () => {
    it('has expected html structure', () => {
      const cmp = shallow(LoadingScreen, { store, localVue })
      expect(cmp.element).toMatchSnapshot()
    })
  })
})