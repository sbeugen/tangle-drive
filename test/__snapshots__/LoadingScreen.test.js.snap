// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingScreen.test.js snapshot has expected html structure 1`] = `
<div
  class="screen-background"
>
  <div
    class="spinner-background"
  >
    <!---->
     
    <br />
     
    <h2>
      Upload Prep.
    </h2>
  </div>
</div>
`;
